### ISSUE TITLE
"thank you" feature

### ISSUE THREAD

**Issue by fuddl** [openstreetmap/openstreetmap-website#875]
wikipedia has this very nice motivating feature. it is a button next to every changeset which is labeled _thanks_. When clicked, the contributor gets a little notification like this:

![Thank you on wikipedia](https://cloud.githubusercontent.com/assets/842548/5806250/bb398876-a017-11e4-808f-930623e1a15b.png)

I suppose its not hard to integrate into osm and its very motivating… at least for me

**Comment by tomhughes** [comment]
Please god no

**Comment by vincentdephily** [comment]
@fuddl just use the existing changeset discussion feature ? It's a few more keypresses, but that makes receiving a "thank you" even more motivating.

**Comment by danstowell** [comment]
I'd like to argue against "Please god no". Such features encourage us towards more of a feeling of community oversight and second-checking of changes, in a positive rather than argumentative manner. It strikes me as healthy for the OSM community.

On the other hand, I think as @vincentdephily says, for now at least we can use the changeset discussion feature to prototype that as a feature of our community. I've already been adding little thankyous here and there, and I think it's worth doing, *without* at this point adding any more technical features.

**Comment by tomhughes** [comment]
So my objection is really only if it's intrusive to the recipient - if it's like a facebook like or twitter favourite or whatever, that just increments a counter somewhere that it there if I choose to look at it then fine.

What I don't want is something in my face that I have to actively dismiss/read/file and that acts more like the incredibly tedious thing that is people +1ing bugs.

**Comment by tmcw** [comment]
This could be annoying, but it also has the potential to be a nice and simple community feature: I think good design can avoid making it intrusive, and @fuddl's sketch looks good.

**Comment by pnorman** [comment]
With changeset comments we offer a mechanism for users to offer constructive positive feedback, which a simple :+1: does not provide. I would much rather see that used, as my experience has been that such feedback helps encourage mappers more than a like button.

**Comment by kerosin** [comment]
In general I would like to see a notification system for things like notes or changesets comments (by now only email-notification is supported which is kind of annoying).

**Comment by brycenesbitt** [comment]
"NextDoor Neighbor" has this feature, and I thought I'd hate it.  As it turns out it's a nice little recognition feature, and I find it motivating.

That said one can already write a thank you note in a changeset, and the mapper will see it.  This proposal could be seen as a "lite" version of that: more interactive than staying silent, less intrusive than commenting on a changeset or messaging the mapper.

Thinking about creating habits (as http://www.nirandfar.com/2012/03/how-to-manufacture-desire.html does) the feedback creates a dopamine hit, a reward for a good edit.

**Comment by Zverik** [comment]
I agree. "Thanks you"s can be integrated in changeset stream monitoring services, so people monitoring their area could thank new mappers easier, without copypasting a template thank you notice. As I told at SotM 2013, there is not enough positive feedback in OSM, so why not make this simple feature.

**Comment by carstenhag** [comment]
I'd also really like this and it's always missing (again and again I want to say thanks, but as the note is closed there's no way)!

**Comment by danieldegroot2** [comment]
See also https://github.com/osmlab/osm-planning/issues/23, #1076, [post by haraldkliems](https://fosstodon.org/@haraldkliems/113283366565954467)

Related #693 / #844, [post by 02JanDal](https://community.openstreetmap.org/t/proposal-add-reactions-votes-to-changesets/106730/38)

Furthermore, the OSM Wiki has added the [Thanks extension](https://www.mediawiki.org/wiki/Extension:Thanks) ([2019-01](https://github.com/openstreetmap/operations/issues/265));
<sub>it is also available for discussion comments as of adding the [DiscussionTools extension](https://www.mediawiki.org/wiki/Extension:DiscussionTools#Optional_dependencies) ([2026-03](https://github.com/openstreetmap/operations/issues/1335))</sub>

**Comment by OpenGreenStreet** [comment]
Is there any hope of this being implemented soon? I would also very much like to have such a function!

https://github.com/openstreetmap/openstreetmap-website/issues/4542
https://commons.wikimedia.org/wiki/File:20240225xl_171048-WP_Give_thanks.jpg

**Comment by jacobwhall** [comment]
I'd love to see a "thank you" feature on the OpenStreetMap site. Some of my most cherished memories within the OSM community have been personal notes thanking each other for stand-out edits and other contributions to our project. I agree with @fuddl that MediaWiki is a great example of this—I've been "thanked" a couple of times for OpenStreetMap Wiki contributions which has also been encouraging.

**Comment by 1ec5** [comment]
Agreed, this feature could help the community maintain a reasonably positive, constructive atmosphere. Users will continue to write out comments and private messages when they have more to say. Another common use of Thanks on MediaWiki is actually to soften the blow of a revert, when the reverter recognizes that the original change was well-intentioned but is no longer a good idea for some reason. It helps to deescalate that kind of situation.[^first]

For clarity, let’s consider the scope of this feature request to be something similar to what MediaWiki has: a more or less private expression of gratitude to the author of a changeset. Public expressions of gratitude, such as emoji reactions, are out of scope. Based on what people say they want this feature for, I think we should focus on the changesets themselves before thinking about any thanks for individual comments on changesets or diary posts.

As expressed in https://github.com/openstreetmap/openstreetmap-website/issues/875#issuecomment-70642998, this feature would need to be unobtrusive to really have the desired motivating effect. This would need to feed into an in-site notification rather than the existing e-mail notifications: #1387.

In addition to notifications, we’d need to track which changesets a given user has already given thanks for. The Thanks button needs to be disabled once you’ve used it on a given changeset. Otherwise, it would seem to be broken, or you might forget you’ve already used it and keep thanking someone for the same thing. To keep this from becoming a tool for badgering someone passively aggressively, we might need to consider some sort of rate limit per thanker, but maybe that can be tail work.

[^first]: I’m sure SteveC has muted [changeset 1](https://www.openstreetmap.org/changeset/1) by now, but a Thanks feature should also mitigate that kind of pile-on to some extent.

**Comment by OpenGreenStreet** [comment]
I am very much in favor of introducing a “thank you” feature. (One drawback of MediaWiki is that, for the user who clicks “thank you,” the system only remembers that they have already done so until the next time they log out. After logging back in, they might accidentally click “thank you” again for the same thing.)

**Comment by Kovoschiz** [comment]
OSMCha has the limitation of only 1 user being able to thumbs-up. Sometimes one may want to thank an attempt, but not thumbs-up for having problems.
